### PR TITLE
Test case for BFD echo-mode

### PR DESCRIPTION
**Pull request by Ghulam-Bahoo**
### Description of PR

Summary:
This PR introduces a new test case in the existing [BFD test plan](https://github.com/sonic-net/sonic-mgmt/pull/11898)  focused on testing the existing BFD echo-mode feature. By testing BFD echo-mode functionality, I aim to ensure its reliability and effectiveness in real-world scenarios. 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Enabling BFD Echo-mode provides an additional layer of fault detection in network environments. This proactive approach helps to minimize network downtime and improve overall network reliability.
#### How did you do it?
Enabling BFD echo-mode across all neighbors in the topology.
#### How did you verify/test it?
I verified the test by running the command:
- show bfd peer 

which will show all the details about bfd neighbors.

### CODE REVIEW

**Comment by ridahanif96** [comment]
@yxieca Hi Ying, can you please assign a reviewer for this testcase.

File: tests/bfd/test_bfd.py [hunk 1]
```diff
@@ -481,3 +490,35 @@ def test_bfd_multihop(request, rand_selected_dut, ptfhost, tbinfo,
         duthost.shell(cmd_buffer, module_ignore_errors=True)
         ptfhost.command('supervisorctl stop bfd_responder')
         ptfhost.file(path=BFD_RESPONDER_SCRIPT_DEST_PATH, state="absent")
+
+
+@pytest.mark.parametrize('ipv6', [False, True], ids=['ipv4', 'ipv6'])
+def test_bfd_echo_mode(request, rand_selected_dut, ptfhost, tbinfo, ipv6):
```
Comment (siqbal1986) [inline tests/bfd/test_bfd.py:496]: Can you please point to me where the actual BFD session is being created.
From what  i understand you are simply setting vtysh -c 'configure terminal' -c 'bfd' -c 'bfd peer {neighbor IP}' -c 'echo-mode'
and then you do 
vtysh -c 'show bfd peer'"
Is the intent of the test simply to test i f the echo mode is configurable.

File: tests/bfd/test_bfd.py [hunk 2]
```diff
@@ -481,3 +490,35 @@ def test_bfd_multihop(request, rand_selected_dut, ptfhost, tbinfo,
...
+        error = False
+        for line in result:
+            if ('Echo transmission interval:' in line) or (
+                    'Echo receive interval:' in line):
+                if 'disabled' in line:
+                    error = True
+        assert error is False
+    finally:
+        # Cleanup: Remove BFD sessions and echo mode configurations
+        remove_bfd_sessions(duthost, neighbor_addrs)
```
Comment (siqbal1986) [inline tests/bfd/test_bfd.py:519]: if no sessions are created why the cleanup.

File: tests/bfd/test_bfd.py (hunk 1 above)
Comment (Ghulam-Bahoo) [reply tests/bfd/test_bfd.py:496]: This command establishes the BFD neighborship first and then enables the echo-mode on the peer. The breakdown of the command is as follows:
-c bfd (to instantiate the bfd instance)
-c bfd peer (neighbor IP)
These two commands enable the bfd session.

-c echo-mode (enabling echo-mode on the peer)
This command is to configure the peer.

Refer to lines 326-328 in bfd_test.py.

File: tests/bfd/test_bfd.py (hunk 2 above)
Comment (Ghulam-Bahoo) [reply tests/bfd/test_bfd.py:519]: BFD sessions are being created first as mentioned in the previous comment. So, cleanup is required for session tear-down.

**Comment by ridahanif96** [comment]
@siqbal1986 Hi, Can you please help expedite this test-case. Target is 202405 release. Thanks in advance for help.

**Review by siqbal1986 — APPROVED** [review]

**Comment by ridahanif96** [comment]
@yxieca can you pls help merge this testcase.

**Comment by dgsudharsan** [comment]
@Ghulam-Bahoo  @siqbal1986 This test uses FRR based bfd and not BFD offload. However for FRR based BFD we need to have frr_mgmt_framework_config set to true
https://github.com/sonic-net/sonic-buildimage/blob/72b2f79c75705ccdfc73f6b2a0ab05c375eff400/dockers/docker-fpm-frr/frr/supervisord/supervisord.conf.j2#L65

I don't see this test case explicitly introducing this configuration. How does this test work?

We see the following error

```
vtysh -c 'configure terminal' -c 'bfd'
bfdd is not running
```

@nhe-NV FYI

**Comment by bingwang-ms** [comment]
@Ghulam-Bahoo Can you please help check?

**Comment by siqbal1986** [comment]
@Ghulam-Bahoo @ridahanif96 Can you please address the issue in this test as mentioned in an earlier comment. This test has been failing from day one. the other choice is to revert this PR.

**Pull request merged**